### PR TITLE
Get Directions

### PR DESCRIPTION
**Pull request by A-N1khil**
- [x] Segregate Location selector and keep it reusable
- [x] Set source and destination UI
- [x] Get directions
- [x] Allow users to set location

### CODE REVIEW

**Review by shivamraj-web — APPROVED** [review]

**Review by sh1v4mr4j — APPROVED** [review]

File: client/src/components/Maps/AddressSetter.jsx [hunk 1]
```diff
@@ -1,90 +1,97 @@
...
+      })
+      .catch((error) => console.error(error))
+      .finally(() => setSearching(false));
+  };
 
   const onSubmitFailed = (errorInfo) => {
-    console.log('Failed:', errorInfo)
-    setShowAlert(true)
-  }
+    console.log("Failed:", errorInfo);
```
Comment (sh1v4mr4j) [inline client/src/components/Maps/AddressSetter.jsx]: can be removed in the future.

File: client/src/components/Maps/Directions.jsx [hunk 2]
```diff
@@ -1,185 +1,119 @@
...
-  // Alerts
-  const [showAlert, setShowAlert] = useState(false);
+  const [source, setSource] = useState();
+  const [destination, setDestination] = useState();
 
-  const [menuResults, setMenuResults] = useState([]);
-  const [selectedMenuId, setSelectedMenuId] = useState("");
-
-  const handleMenuClick = (e) => {
-    console.log("onClick");
```
Comment (sh1v4mr4j) [inline client/src/components/Maps/Directions.jsx]: I guess you will remove console logs in future or can comment for now and use in dev before merging?

File: client/src/components/Maps/AddressSetter.jsx (hunk 1 above)
Comment (A-N1khil) [reply client/src/components/Maps/AddressSetter.jsx]: Not up for review atm. Just placeholder texts

Comment (A-N1khil) [reply client/src/components/Maps/AddressSetter.jsx]: Done

File: client/src/components/Maps/Directions.jsx (hunk 2 above)
Comment (A-N1khil) [reply client/src/components/Maps/Directions.jsx]: Done

File: client/src/components/Maps/AddressSelector.jsx [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+import {
+  updatePatientLocation,
+  updateDoctorLocation,
+} from "../../api/services/locationService";
+
+const AddressSelector = () => {
+  const { user } = useUserContext();
+  const [messageApi, contextHolder] = message.useMessage();
+
+  const [open, setOpen] = useState(true); // fixme Change to false
```
Comment (sh1v4mr4j) [inline client/src/components/Maps/AddressSelector.jsx]: fix her now?

**Review by sh1v4mr4j — APPROVED** [review]
Looks good, can merge

File: client/src/components/Maps/AddressSelector.jsx (hunk 3 above)
Comment (A-N1khil) [reply client/src/components/Maps/AddressSelector.jsx]: Fixed

**Comment by snigdhansu** [comment]
Can you please paste screenshot of changes and update screenshot folder accordingly

File: server/app/services/patient_service.py [hunk 4]
```diff
@@ -68,7 +68,7 @@ async def scheduleAppointment(self, email: str, appointmentDetails: Appointment)
         response = {"message": "Appointment updated successfully", "currentAppointment": updated_appointments}
         return 201, response
     
-    async def set_address_for_patient(self, patient_email: str, address: Location):
+    async def set_location_for_patient(self, patient_email: str, address: Location):
```
Comment (snigdhansu) [inline server/app/services/patient_service.py:71]: can we keep this consistent, during function call we use location, and here we use address as args variable name

**Review by snigdhansu — APPROVED** [review]

**Pull request merged**